### PR TITLE
Fileattachment fixes

### LINKED ISSUE #847: File dropper is calling bb-file-drop-change before file is selected
### Expected behavior

I expect the `bb-file-drop-change` function to only get called after the user has either dropped a file it or after the user has clicked it to open the browse menu and selected an file.
### Actual behavior

Dropping a file is working fine, however, when clicking the component to open the systems browse menu it seems to be calling the file dropped function before the user has actually clicked on a file to upload.
### Steps to reproduce

Clicking the component for the first time works fine, the function don't get called until a file is selected. Clicking the component subsequently after the first click calls the function when the browse menu opens.
Interestingly, if you then drop a file on it and then click it again, it goes back to functioning as it should as if it's the first time you've clicked it. It seems to only be doing this after clicking the component at least once and stops doing it after dropping a file.
### Plunker

I've altered the default plunker a little bit to add console logs so you can see this happening in the conole.
http://plnkr.co/edit/1TuFse7vGuTzI8vucZ7E?p=preview

### PR DESCRIPTION
**Pull request by Blackbaud-PatrickOFriel**
#847 #857 #844

### CODE REVIEW

File: js/sky/src/fileattachments/docs/demo.html [hunk 1]
```diff
@@ -17,12 +17,12 @@
             </bb-alert>
             <div ng-repeat="item in fileAttachmentDemo.allItems" bb-file-item="item" bb-file-item-delete="fileAttachmentDemo.deleteAttachment(item)" style="margin-bottom: 10px">
                 <div class="form-group">
-                    <label>Name for this attachment</label>
-                    <input type="text" class="form-control" />
+                    <label for="filename" >Name for this attachment</label>
+                    <input id="filename" type="text" class="form-control" />
```
Comment (Blackbaud-BobbyEarl) [inline js/sky/src/fileattachments/docs/demo.html]: Should we append the `$index` here in order to make sure they're unique inside the `ng-repeat`?

Comment (Blackbaud-PatrickOFriel) [reply js/sky/src/fileattachments/docs/demo.html]: Yes I'll do that

Comment (Blackbaud-PatrickOFriel) [reply js/sky/src/fileattachments/docs/demo.html]: Updated

**Comment by Blackbaud-BobbyEarl** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/blackbaud/skyux/pull-request/860/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=blackbaud/skyux)

**Pull request merged**